### DISCUSSION TITLE
Where I can access the generate report?

### DISCUSSION THREAD

**Discussion by CosteaIonut89** [pytest-dev/pytest-html#529] [Q&A]
Hello, I want to check the function where the report is generated because I want to edit if the test passes the report that should not be generated.

**Comment by BeyondEvil** [comment]
Not sure I follow.

Could you clarify? Maybe with a concrete example?

**Reply by CosteaIonut89** [reply]
I am in the package pytest-html -> plugin.py and there are some functions that return the pass test cases in the report for example `    def append_passed(self, report):
        if report.when == "call":
            if hasattr(report, "wasxfail"):
                self.xpassed += 1
                self._appendrow("XPassed", report)
            else:
                self.passed += 1
                self._appendrow("Passed", report)`
Here if I edit something the pass will not be generated but if the test case is still passed the report-HTML is still displayed so my question will be in that file plugin.py I don't understand where the report is generated for the pass status for a test case. I am using IDE PyCharm so there I found this function and I did the edits.
If I am wrong with something I am sorry about that. Thank you in advance!

**Reply by BeyondEvil** [reply]
So, with v3 and older the report is only generated (written to disk) once _all_ tests have run. 

In that case, you're probably looking for `def _generate_report`.

In the (not yet released) v4, the report is generated and updated continuously.

**Reply by CosteaIonut89** [reply]
I search for this function because of these idea [(https://github.com/pytest-dev/pytest-html/discussions/528)] I was thinking I am possible to modify the plugin.py where the report is generated to add or remove the condition if the test cases are passed.
